### ISSUE TITLE
Feature Request: Admins can always edit their comments

### ISSUE THREAD

**Issue by GiovanH** [umputun/remark42#946]
It's often helpful for a moderator to pin a comment at the top of the feed with relevant information, but currently moderators are always bound by the EditInterval and the "can't edit replied-to posts" rule. It would be nice if there were an option to always allow moderators to edit their posts.

**Comment by umputun** [comment]
Well, this can be done technically, but I don't think allowing admins to always edit comments is such a good idea. The conversation will make no sense if one of the participants can edit comments people already replied to.

As a workaround for pinned comment update, I can suggest unpinning the current one, make another comment and pin it.

**Comment by GiovanH** [comment]
That works for updating the pinned comment, but it doesn't maintain continuity if people have *replied* to the pinned comment. 

I guess built into this is the assumption that moderators can be trusted to maintain basic continuity (simply appending new info at the bottom of the comment, for instance, rather than changing the meaning of the whole thing). I wouldn't extend that trust to regular anonymous users, for obvious reasons.

**Comment by umputun** [comment]
> simply appending new info at the bottom of the comment

this is, actually, an interesting idea. maybe allowing admins to append-only and mark it automatically as "updated at ..." will be a good compromise.

**Comment by GiovanH** [comment]
Sure, but that's just one sensible use case of editing posts. There are plenty of other sensible ways someone can sensibly edit a post: crossing out old information, for instance, or closing a call for response because the date has passed.

You wouldn't want to build a whole feature set around just the ability to append *instead* of the edit setting I described. Especially if it's a setting that only applies to moderators, and can be configured on and off.

**Comment by umputun** [comment]
My fundamental problem with this is breaking the basic idea of "enforced trust" remark42 is trying to implement. To me, the ability to change comments by moderators essentially the same as ability to edit any user's comment by moderators, because the final result could be very similar - the total confusion and lost of the original context. Sure, moderators are trusty individual and they even allowed to remove comments and block users. However removing comments or even all thread still preserves comment's immutability in some sense.

Anyway, I get the idea and trying to understand the use case better. To me the whole "pin a comment at the top of the feed with relevant information" with "update it latter" sounds like an attempt to leverage remark42 for something like forum, where everything just comments of some sort. What is the particular use case you have in mind? Why moderator communicate some essential information via pinned comment and not with whatever those comments attached to (i.e. blog post, message, etc)?

**Comment by GiovanH** [comment]
In my most recent use case, a team of trolls was trying to use my comments section to dox someone I mentioned in a blog article. I wanted to pin a comment making it clear that this was not allowed and offending comments would be deleted. I also disabled some forms of authentication, but only temporarily. A pinned comment is a good way to have an announcement that's *about the comments section*, rather than just trailing content at the end of the main article.

I think the idea of "enforced trust" is a good one, but ultimately it has to be backed by the site owner. If an administrator were truly malicious they can cause huge problems already; they can delete posts or even edit the data store to fabricate whatever content they wanted. And that assumes they're really running vanilla Remark42, instead of a fork that implements this feature. 

**The administrator has to be a fundamental point of trust.** Disallowing administrators to edit posts on principle just obfuscates the trust problem slightly; it's not any kind of real security. With that in mind, remark42 should give administrators the tools they need to properly moderate their websites, instead of trying software that defends itself from its own administrators.

**Comment by umputun** [comment]
> If an administrator were truly malicious they can cause huge problems already; they can delete posts or even edit the data store to fabricate whatever content they wanted. And that assumes they're really running vanilla Remark42, instead of a fork that implements this feature.

I think we are talking about two different things here, or at least two unrelated things got mixed up in the process. The administrator in remark42 is what called a moderator in other systems. This doesn't have to be the same person who installed the droplet/server and has all keys from the kingdom. Only the "real admin" (or server owner) can do everything on this server. Btw, I have seen multiple setups where remark's admin just this - moderator, without any superpowers to store or fabricate whatever content they wanted or install a customer version of remark42 on the server.

> The administrator has to be a fundamental point of trust

I get it, however back to my previous example - we don't have different levels of admins/moderators/etc, so this is some sort of all-or-none balance. Your position - it should be all and I appreciate it. Mine - as little as possible to do the job.

Anyhow, I see what you mean and the use case makes sense. Having this ability as an option (likely off by default) will keep the decision to the site owner and I can live with it. 

Technically implementing this special power for admin users is somewhat relatively simple on the backend side. However, I'm not sure if the frontend will be that easy. @akellbl4 do you recall, for example, how this countdown thingy done on your side? I don't think the backend can control it on the per-comment level it and it probably means a special logic in UI to drop this countdown for admins and smth similar for hiding/showing "edit" link.

**Comment by umputun** [comment]
the backend support in place. not sure how this will play with the current frontend but may be worth a try.

to support it fully on UI side we should eliminate the logic with countdown for admin if this mode enabled. I have added `admin_edit` value to the config call

**Comment by akellbl4** [comment]
@umputun you are right. We should show the edit button always without the countdown. Should be easy to do.

**Comment by zy9306** [comment]
Anyone familiar with the front-end can help with this?

**Comment by akellbl4** [comment]
@zy9306 I will

**Comment by geerlingguy** [comment]
Ditto on wanting to have this feature; on my blog I'm the only admin, and there are many times where I need to adjust one of my comments (usually fat-fingering a post with a bad link/formatting) after posting it.

I am also very careful about preserving the history of a comment, too; the main thing would be to maybe add a little `(edited)` any time a comment is edited (even if it's within the first window where edits are allowed for everyone).

**Comment by paskal** [comment]
This was implemented in #2001, released in [v1.16.0](https://github.com/umputun/remark42/releases/tag/v1.16.0). With `ADMIN_EDIT=true`, admins can edit their own comments after the normal edit window expires, including comments with replies. #2205 tracks the separate countdown display defect on freshly posted admin comments.
